fiftyone/utils/labels.py

```diff
@@ -341,3 +343,115 @@ def classification_to_detections(sample_collection, in_field, out_field):
             image[out_field] = fol.Detections(detections=[detection])
 
         sample.save()
+
+
+def filter_keypoints(sample_collection, field, expr=None, labels=None):
```

Comment:
Shouldn't this proivde an `only_matches` option? If the confidence filter does not filter empty keypoints than it is better to do this all client side in the App.

Reply:
Hmm yeah that does make sense. If a filter omits all points in a `Keypoint`, remove the entire object.

I think I can add that

Reply:
@benjaminpkane ok added the `only_matches` flag.

I'm thinking about converting this utility to a proper `FilterKeypoints(ViewStage)` and `SampleCollection.filter_keypoints()` view stage. Any objections?

Reply:
I might prefer `FilterLabels` to support keypoints automatically. Perhaps in addition to `FilterKeypoints`. `FilterKeypoints` is somewhat specific but I suppose thats fine for now.

Reply:
I thought about merging this functionality into `FilterLabels`, but the problem is that will require introducing new optional kwargs, say `points_filter` and `points_labels`, that are only applicable when the field being filtered happens to be of type `Keypoint(s)`.

Reply:
Ok. Cause to refactor `Keypoint` in the future